sdcm/wait.py

```diff
@@ -16,11 +16,46 @@
 """
 import time
 import logging
+from avocado.utils import wait
 
-LOGGER = logging.getLogger('avocado.test')
+LOGGER = logging.getLogger('sdcm.wait')
 
 
-def wait_for(func, step=1, text=None, *args, **kwargs):
+class WaitTimeoutExpired(Exception):
+    pass
+
+
+def wait_for(func, step=1, text=None, timeout=None, throw_exc=False, **kwargs):
+    """
+    Wrapper function to wait with timeout option.
+    If timeout received, avocado 'wait_for' method will be used.
+    Otherwise the below function will be called.
+
+    :param func: Function to evaluate.
+    :param step: Time to sleep between attempts in seconds
+    :param text: Text to print while waiting, for debug purposes
+    :param timeout: Timeout in seconds
+    :param throw_exc: Raise exception if timeout expired, but func result is not True
+    :param kwargs: Keyword arguments to func
```

Comment:
:param kwargs: Keyword arguments to func, contains both converted non-keyword arguments and original keyword arguments. Example: foo(arg1, arg2=None) -> wait_for(foo, arg1='v1', arg2='v2')

Reply:
I removed an optional positional args, so you can only pass keyword args.
This is an example from the existing code(see cluster.py):
wait.wait_for(func=self.cfstat_reached_threshold, step=10,
                      **text=text, key=key, threshold=threshold**)